kubernetes/plausible.yaml

```diff
@@ -0,0 +1,101 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: plausible
+  labels:
+    app.kubernetes.io/name: plausible
+    app.kubernetes.io/component: server
+spec:
+  type: LoadBalancer
+  ports:
+    - name: http
+      port: 8000
+      targetPort: 8000
+      protocol: TCP
+  selector:
+    app.kubernetes.io/name: plausible
+    app.kubernetes.io/component: server
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: plausible
+  labels:
+    app.kubernetes.io/name: plausible
+    app.kubernetes.io/component: server
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: plausible
+      app.kubernetes.io/component: server
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: plausible
+        app.kubernetes.io/component: server
+    spec:
+      restartPolicy: Always
+      containers:
+        - name: plausible
+          image: plausible/analytics:latest
+          imagePullPolicy: Always
+          args:
+            - /bin/sh
+            - -c
+            - sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh db init-admin && /entrypoint.sh run
```

Comment:
It might be a good idea to split this up and use an [init container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) for the `sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh db init-admin` part. The actual container only executes `/entrypoint.sh run`.

See [my deployment](https://github.com/schnerring/infrastructure/blob/main/plausible.tf#L231-L269) (mind it's written in Terraform).